app/models/forms/export.rb

```diff
@@ -48,5 +198,48 @@ def name(qing)
         "Group"
       end
     end
+
+    def to_number(value)
+      return if value.blank?
+      (value.to_f % 1).positive? ? value.to_f : value.to_i
+    end
+
+    # Takes an array of conditions and outputs a single string
+    # concatenates by either "and" or "or" depending on form settings
+    def conditions_to_xls(conditions, true_if)
+      relevant_to_push = ""
+      concatenator = true_if == "all_met" ? "and" : "or"
+
+      # how many conditions?
+      dc_length = conditions.length
+
+      conditions.each_with_index do |dc, i|
+        # prep left side of expression
+        left_qing = Questioning.find(dc.left_qing_id)
+        left_to_push = "${#{left_qing.full_dotted_rank}_#{left_qing.code}}"
+
+        # prep right side of expression
+        if dc.right_side_is_qing?
+          right_qing = Questioning.find(dc.right_qing_id)
+          right_to_push = "${#{right_qing.full_dotted_rank}_#{right_qing.code}}"
+        elsif Float(dc.value, exception: false).nil? # it's not a number
+          # to respect XLSform rules, surround with single quotes unless it's a number
+          right_to_push = "'#{dc.value}'"
+        else
+          right_to_push = dc.value.to_s
+        end
+
+        op = OPERATIONS[dc.op]
+
+        # omit the concatenator on the last condition only
+        relevant_to_push = if i + 1 == dc_length
+                             "#{relevant_to_push}#{left_to_push} #{op} #{right_to_push}"
+                           else
+                             "#{relevant_to_push}#{left_to_push} #{op} #{right_to_push} #{concatenator} "
```

Comment:
to simplify the logic here, i recommend making `relevant_to_push` an array of strings instead of one big string, and then return `relevant_to_push.join(concatenator)` at the very end